### PR TITLE
How to create a font chooser App

### LINKED ISSUE #4082: How to create a font chooser app using python
#### Proposed article introduction
Font chooser app is a simple application that lets you browse the installed fonts on your computer and then narrow down that selection till you find the perfect font for your design project or write-up. By Clicking on the sample text to change it to your chosen desirable text. we will build this GUI app using tkinter in python.

#### Key takeaways 
the reader should be able to 
- know all GUI methods
- the role and function of tkinter as a GUI method and in data science.
- how to build a font chooser app.
#### Article quality
How is your article unique? there are few or no articles publish on this subject and it is going to be more detailed and easy to understand for a reader who is a beginner in a programming language

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by majiezra**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4082 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@majiezra Please check the comments below.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+By defining the `user_text` variable, we can add the textbox. We do this by putting the defined variable in a frame and utilising our font to make the font size dynamic.
+We utilize the `.grid` function to prevent the font app from resizing abnormally, and we set the row and column weights to 1 to ensure that it evenly distributes any additional space in the row and column.
+We create a Listbox by setting the `my_listbox` variable to a listbox with a single-mode selection, which allows us to choose only one font at a time while still specifying the width size.
+
+```python
+for f in font.families():
+my_listbox.insert('end', f)
+my_listbox.bind('<ButtonRelease-1>', font_chooser)
+root.mainloop()
+```
```
[the remark this answers is not in the document]
Comment (majiezra) [reply content/articles/how-to-create-font-chooser-app/index.md]: Done @geoffrey45

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+```python
+pip install tk
+```
+
+### The graphical user interface
+The graphical user interface is the most common type of user interfaces. One can interact with these interfaces by pointing and clicking on graphics or icons with a mouse, trackpad, or other peripherals.
+
+#### Advantages of the graphical user interface
+- It is simple to use.
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:50]: 1st statement in this list:

The GUI is not simple to use. It's simpler than the CLI. Please update your statement to reflect this.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+- There is no need to memorize the command lists as it is easy to visualize and remember.
+- It allows more than one application and program to run at the same time.
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:52]: The 3rd point in this list is not fully true. The CLI can also run multiple applications simultaneously using background processes.

Please update your statement to use well defined words like "multitasking", etc.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+- It provides many exceptional support services.
+- It is self-explanatory.
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:54]: The 5th statement in this list:

The meaning of the word "self-explanatory" is "[capable of being understood without explanation](https://www.merriam-webster.com/dictionary/self-explanatory#:~:text=:%20explaining%20itself%20:%20capable%20of%20being%20understood%20without%20explanation)". The self-explanatory-ness of a GUI depends on how simple or complex it is.

Please remove or update this statement  to enhance the correctness.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+To start, first, we will create a root widget and this root widget is a window with a title bar. This widget has to be created first, and it is called the root widget, after creating the root widget, the programmer can then create other widgets.
+
+Then we give our font app a name (I used `maji.ico`. You can use your own name if you want to customize the app.). Then we set the size of the application.
+
+```python
+def font_chooser(a):
+user_font.config(
+ family=my_listbox.get(my_listbox.curselection()))
+```
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:85]: Properly format your code blocks.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+We create a `Listbox` by setting the `my_listbox` variable to a listbox with a single-mode selection, which allows us to choose only one font at a time while still specifying the width size.
+
+```python
+for f in font.families():
+  my_listbox.insert('end', f)
+
+
+my_listbox.bind('<ButtonRelease-1>', font_chooser)
+root.mainloop()
+```
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:125]: Properly format your code blocks.

File: content/articles/how-to-create-font-chooser-app/index.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+And the font chooser interface will look like this:
+![fontapp image](/engineering-education/how-to-create-font-chooser-application/fontchooser.png)
+
+### Conclusion
+In this tutorial, we reviewed the role of Tkinter in data science and built a font chooser application.
+
+### Reference
+[Python GUI programming with Tkinter](https://www.perlego.com/book/721869/python-gui-programming-with-tkinter-pdf)
+
```
Comment (geoffrey45) [inline content/articles/how-to-create-font-chooser-app/index.md:139]: Please include the author of this book in this reference. ie. `Python GUI programming with Tkinter - By author name`

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
Hello @majiezra 👋

Your article looks good. Just a few issues.

You have spent a large chunk of the article talking about the GUI and Tkinter for data analysis. Yet, your article aims at guiding the reader on building a font chooser application using Tkinter.

I feel like you should be talking more about fonts, how they work, or something related to fonts and what your article aims at helping the reader achieve.

Please don't stray away from the topic of discussion.

If you feel like the history of the GUI and how Tkinter can be used for data analysis is something the readers need to know, please leave external links to published material containing the information.

Feel free to dedicate a whole new article to this topics.

Please remove this unrelated areas, and replace them with content that is related to what your article is talking about.

**Comment by hectorkambow** [comment]
@majiezra 

As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
All revisions and edits should be completed on your own forked repo (as to not build up a backlog in the queue). 

cc @geoffrey45 
If during the tech review we discover the PR still needs more work - we can close this PR and have the author re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself.

**Pull request closed without merging**